include/bout/solver.hxx

```diff
@@ -268,10 +268,17 @@ class Solver {
     throw BoutException("resetInternalFields not supported by this Solver");}
 
   // Solver status. Optional functions used to query the solver
-  virtual int n2Dvars() const {return f2d.size();}  ///< Number of 2D variables. Vectors count as 3
-  virtual int n3Dvars() const {return f3d.size();}  ///< Number of 3D variables. Vectors count as 3
-  
-  int rhs_ncalls,rhs_ncalls_e,rhs_ncalls_i; ///< Number of calls to the RHS function
+  /// Number of 2D variables. Vectors count as 3
+  virtual int n2Dvars() const {return f2d.size();}
+  /// Number of 3D variables. Vectors count as 3
+  virtual int n3Dvars() const {return f3d.size();}
+
+  /// Get and reset the number of calls to the RHS function
+  int resetRHSCounter();
```

Comment:
I'm not sure about using a single method to both get the value and reset it at the same time. It's useful for how it's currently used, but what if a monitor just wants to get the number of RHS calls without changing it?

Reply:
Agree we can add `getRHScounter` or similar if there is a demand for it.